extensions/amp-story/validator-amp-story.protoascii

```diff
@@ -803,6 +804,48 @@ tags: {  # <amp-story-page-attachment> with no href
     value: "light"
   }
 }
+tags: {  # <amp-story-animation> <script>
+  html_format: AMP
+  tag_name: "SCRIPT"
+  spec_name: "amp-story-animation json script"
+  mandatory_parent: "AMP-STORY-ANIMATION"
+  satisfies: "amp-story-animation json script"
+  attrs: {
+    name: "type"
+    mandatory: true
+    value_casei: "application/json"
+    dispatch_key: NAME_VALUE_PARENT_DISPATCH
+  }
+  attr_lists: "nonce-attr"
+  cdata: {
+    blacklisted_cdata_regex: {
+      regex: "<!--"
+      error_message: "html comments"
+    }
+  }
+}
+tags: {  # <amp-story-animation>
+  # No amp_layout since this doesn't have a custom element implementation.
+  html_format: AMP
+  tag_name: "AMP-STORY-ANIMATION"
+  requires_extension: "amp-story"
+  requires: "amp-story-animation json script"
```

Comment:
I didn't know you could somehow validate an element child through this `requires`, that's very cool. Can you specify several `requires` statements?
cc @newmuis FYI

Reply:
You can specify multiple `requires`:

https://github.com/ampproject/amphtml/blob/6f30fdad6715ffced172d5dede2276291f3de71e/validator/validator.proto#L603

Any particular ideas?

Reply:
The `satisfies` / `requires` pairs are document global. In other words, they don't express any `parent` or `child` requirements. If anything on the document matches something with a `requires` bit, something else in the document will need to match the same `satisfies` bit. This was originally implemented for verifying extensions were loaded with the `<amp-foo>` element was present, but that system has evolved some.